extensions/map-layers/src/ui/widget/MapLayerManager.tsx

```diff
@@ -64,6 +64,10 @@ function getMapLayerSettingsFromViewport(viewport: Viewport, getBackgroundMap: b
   const displayStyleLayers = (getBackgroundMap ? displayStyle.backgroundMapLayers : displayStyle.overlayMapLayers);
   for (let layerIdx = 0; layerIdx < displayStyleLayers.length; layerIdx++) {
     const layerSettings = displayStyleLayers[layerIdx];
+    if (layerSettings instanceof ModelMapLayerSettings) {
```

Comment:
The maplayers widget depends heavily on `displaystylestate.findMapLayerIndexByNameAndUrl` to interact with the display style..  The URL is important because name is not unique across maplayers. Problem is, model map layer dont have an URL, which make it impossible to lookup a maplayer index based on its definition.

Maybe its time to introduce a unique identifier for maplayer ?

Reply:
I guess that would work too, although caller will have check what type of maplayer he currently hold and pass the appropriate parameters.

Reply:
I'll add a method on baseClass that returns the layer source (URL or ModelID based on type) and then rename methods to findMapLayerByNameAndSource....